detox/src/devices/AppleSimUtils.js

```diff
@@ -44,25 +49,27 @@ class AppleSimUtils {
     return device;
   }
 
-  async waitForDeviceState(udid, state) {
-    let device;
-    await retry({ retries: 10, interval: 1000 }, async () => {
-      device = await this.findDeviceByUDID(udid);
-      if (!_.isEqual(device.state, state)) {
-        throw new Error(`device is in state '${device.state}'`);
-      }
-    });
-    return device;
+  async boot(udid) {
+    if (!await this.isBooted(udid)) {
+      await this._bootDeviceByXcodeVersion(udid);
+    }
   }
 
-  async boot(udid) {
+  async isBooted(udid) {
     const device = await this.findDeviceByUDID(udid);
-    if (_.isEqual(device.state, 'Booted') || _.isEqual(device.state, 'Booting')) {
-      return false;
+    return (_.isEqual(device.state, 'Booted') || _.isEqual(device.state, 'Booting'));
+  }
+
+  async create(type) {
+    const result = await this._execSimctl({ cmd: `list runtimes -j` });
+    const stdout = _.get(result, 'stdout');
+    const runtimes = JSON.parse(stdout);
+    const newestRuntime = _.maxBy(runtimes.runtimes, r => Number(r.version));
+    if (newestRuntime) {
+      console.log('Creating simulator', `create "${type}" "${type}" "${newestRuntime.identifier}"`)
+      await this._execSimctl({cmd: `create "${type}" "${type}" "${newestRuntime.identifier}"`});
```

Comment:
name should not be the type:
```js
const name = `detox_${type}_${count}`
await this._execSimctl({cmd: `create "${name}" "${type}" "${newestRuntime.identifier}"`});
```

Now, when you change the name, you won't be able to query `applesimutils` by name anymore. @LeoNatan added query byType and byId. please check the implementation in:
https://github.com/wix/detox/compare/parallelSupport#diff-2ec04c0d120ceb57a189afbc6a59a9e8R27
https://github.com/wix/detox/compare/parallelSupport#diff-2ec04c0d120ceb57a189afbc6a59a9e8R39